### PR TITLE
Porting nstack to canonical/microkelvin

### PR DESCRIPTION
**Pull request by krl**
Use canonical/microkelvin instead of old kelvin, draft until upstream releases.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -7,292 +7,314 @@
...
 {
-    fn persist(&mut self, sink: &mut Sink<H>) -> io::Result<()> {
-        for handle in self.0.iter_mut() {
-            handle.persist(sink)?;
+    type Leaf = T;
+    type Annotation = A;
+
+    fn annotation(&self) -> Self::Annotation {
+        match self {
+            NStack::Leaf([None, None, None, None]) => A::identity(),
```
Comment (vlopes11) [inline src/lib.rs:41]: We are assuming a constant arity, and it should be defined by `hades252::WIDTH`

File: src/lib.rs [hunk 2]
```diff
@@ -7,292 +7,314 @@
...
+            NStack::Node([Some(a), Some(b), Some(c), Some(d)]) => {
+                let ab = a.annotation().clone().op(b.annotation());
+                let cd = c.annotation().clone().op(d.annotation());
+                ab.op(&cd)
+            }
+            NStack::Node(_) => unreachable!("Invalid node structure"),
+        }
+    }
+
+    fn child(&self, ofs: usize) -> Child<Self, S> {
```
Comment (vlopes11) [inline src/lib.rs:73]: This implementation purpose is unclear

File: src/lib.rs [hunk 3]
```diff
@@ -7,292 +7,314 @@
...
+    A: Canon<S> + Annotation<T>,
+    S: Store,
 {
-    fn persist(&mut self, sink: &mut Sink<H>) -> io::Result<()> {
-        for handle in self.0.iter_mut() {
-            handle.persist(sink)?;
+    type Leaf = T;
+    type Annotation = A;
+
+    fn annotation(&self) -> Self::Annotation {
```
Comment (vlopes11) [inline src/lib.rs:39]: This implementation purpose is unclear

File: src/lib.rs [hunk 4]
```diff
@@ -7,292 +7,314 @@
...
-    type Annotation = A;
-
-    fn children(&self) -> &[Handle<Self, H>] {
-        &self.0
-    }
-
-    fn children_mut(&mut self) -> &mut [Handle<Self, H>] {
-        &mut self.0
+    fn default() -> Self {
+        NStack::Leaf([None, None, None, None])
```
Comment (vlopes11) [inline src/lib.rs:109]: Constant arity, check above

File: src/lib.rs (hunk 1 above)
Comment (krl) [reply src/lib.rs:41]: I disagree, this PR is about porting nstack to microkelvin, it's not about changing/discussing the arity or adding dependencies.

**Review by vlopes11 — APPROVED** [review]
LGTM, lets keep the questions above in mind

File: src/lib.rs (hunk 1 above)
Comment (vlopes11) [reply src/lib.rs:41]: Why do we have 4 `None`?

**Pull request merged**